### DISCUSSION TITLE
Is it possible to read props-initialized store outside of the React Component

### DISCUSSION THREAD

**Discussion by rockcs1992** [pmndrs/zustand#1995] [Q&A]
Hi, first thanks for the excellent project! It helped me solve a lot of problems.

I'm trying to slowly migrate a legacy project to modern stack and ran into a scenario where I need to:
1. The store needs to be initialized with a prop. I saw there is already [documentation ](https://docs.pmnd.rs/zustand/guides/initialize-state-with-props)about that.
2. The store needs to be accessed by its consumers (React components).
3. The store needs to be readable/writable by something outside of the React component tree as well. 

I know I can achieve 1 and 2 together (pretty obvious using the doc above), as well as 2 and 3 together (initializing store with `create`, and then call `getState` and `setState`). But didn't find a way to do all of the three at the same time from the doc. So I'm wondering if this is even possible. If so then are there any examples around that.

**Comment by dai-shi** [comment]
If you want to access the store outside of React, you want to define it globally. In that case, you can _initialize_ the store with props using `useEffect`. (There's a delay, but unavoidable.)
